### PR TITLE
Improve documentation, incl automation statuses

### PR DESCRIPTION
**Pull request by danpetry**
This PR updates the release specs documentation to:

- make it easier to see at-a-glance what tests are present
- make it easier to determine what the automation status of the tests are, and so how we can improve automation.

The changes are:
- Put an overview table in the root folder's README.md
- Clarified the documentation in the root folder's README.md
- Included test overview tables in the individual tests' README.md, including the automation status and difficulty of further automation for all tests
- Renamed all the individual tests' spec documents in all subfolders to README.md, so that it displays in github when clicking into the folders.

There are two designations for automatibility: "Easy" and "hard". This is very broad, on purpose! I don't think any more detail than this is useful. It's meant to give some guidance as to where people can direct their efforts, if they want to automate something.

### How to test

Read!

### CODE REVIEW

File: 04-single-hop-6lowpan-icmp/README.md [hunk 1]
```diff
@@ -1,4 +1,19 @@
-## Goal: Check single-hop ICMP connectivity over IPv6
+Summary
+=======
+
+These tests perform single-hop 6LoWPAN ping tests between nodes in a number of scenarios.
+
+| Task no | Description                                                       | Automated? | Ease of automation |
+|---------|-------------------------------------------------------------------|------------|--------------------|
+| 1       | ICMPv6 link local echo on IoT-Lab M3                              | No         | Hard               |
```
Comment (miri64) [inline 04-single-hop-6lowpan-icmp/README.md]: There is automation for 1-4 and 7 in #79 so I'd say the automation is there and it is not hard ;-)

File: 02-tests/README.md [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
-and **automated** tests for one board, not the manual tests.
+Summary
+=======
+
+These tests compile all applications and tests, and automatically run all tests
+that only require the board, for a selection of individual boards.
+
+| Task no | Description                                | Automated? | Ease of automation |
+|---------|--------------------------------------------|------------|--------------------|
+| 1       | Compile and run tests on native            | Yes        | N/A                |
```
Comment (miri64) [inline 02-tests/README.md]: Not all tests are automated

File: 02-tests/README.md [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+| 2       | Compile and run tests on IoT-Lab M3        | Partly     | Hard               |
```
Comment (miri64) [inline 02-tests/README.md]: Why do you assess it as hard when native is already automated?

File: 05-single-hop-route/README.md [hunk 4]
```diff
@@ -1,4 +1,14 @@
...
+=======
+
+These tests check the static routing of ICMPv6 packets over a single hop.
+
+| Task no | Description                                                | Automated? | Ease of automation |
+|---------|------------------------------------------------------------|------------|--------------------|
+| 1       | ICMPv6 echo unicast addresess on native (default route)    | No         | Easy               |
+| 2       | ICMPv6 echo unicast addresess on iotlab-m3 (default route) | No         | Easy               |
+| 3       | ICMPv6 echo unicast addresess on native (specific route)   | No         | Easy               |
+| 4       | ICMPv6 echo unicast addresess on iotlab-m3 (static route)  | No         | Easy               |
```
Comment (miri64) [inline 05-single-hop-route/README.md]: Single hop link local is hard but single hop with routes is easy *raises eyebrow*?

**Comment by miri64** [comment]
Another thing: Those tables might be helpful on a way to an automated approach, however, what is when the automation is in place? Do we remove them?

**Comment by danpetry** [comment]
Well, ideally the testing should be being continuously improved, including the automation. So once all this is automated, which I guess would be a while, there will be more tests to write, and automate...

In other words, it's intended to be a means of tracking the automation status of the tests, ongoing.

**Comment by miri64** [comment]
> Well, ideally the testing should be being continuously improved, including the automation. So once all this is automated, which I guess would be a while, there will be more tests to write, and automate...
> 
> In other words, it's intended to be a means of tracking the automation status of the tests, ongoing.

Mhhh I see these documents also as information for users what has been tested during the release cycle. So is it a good idea to mix this with WIP information on the tests itself?

**Comment by danpetry** [comment]
Fair point. I could separate the two somehow? (`automation.md`?)

**Comment by danpetry** [comment]
Is it worth also mentioning Jose's open PR? https://github.com/RIOT-OS/Release-Specs/pull/79

**Comment by danpetry** [comment]
Any more comments on these? I could, perhaps, address the general structure of the submission if you have comments on that (e.g. move automation info into their own document, which I think would be good for highlighting them separately)? And then we could go through the "automation difficulty" info offline?

File: README.md [hunk 5]
```diff
@@ -1,35 +1,53 @@
...
+| 09-coap                       | Send and receive of typical CoAP messages                        |
+| 10-icmpv6-error               | Single-hop ICMP connectivity over IPv6 using native              |
+| 11-lorawan                    | LoRaWAN networking                                               |
+| 99-compile-and-test-one-board | Extension - Test 02 for as many extra boards as possible         |
+
+Currently, tests 01, 02 and 09 are mostly automated, whereas the networking
+tests are not. A number of the networking tests can be performed on native, and
+are good targets for automation.
+
+Folder structure
```
Comment (miri64) [inline README.md]: I know the doc said folder previously as well, but these are actually [directories](https://devblogs.microsoft.com/oldnewthing/?p=11473)

Comment (miri64) [reply README.md]: > actually [directories](https://devblogs.microsoft.com/oldnewthing/?p=11473)

Actually actually they are git tree objects, but those are just a git-internal representation of the filesystem concept of directories ;-).

**Comment by danpetry** [comment]
> addition of automation statuses (in which I personally I don't see much added value, especially since they seem pretty arbitrary)

Upon further reflection, I think this information would be better in an issue? Because it's supposed to say, "here's the automation improvements we could make, pick one and improve if you like". which is what is said by issues.
Was thinking an issue to collect automation improvements and one to collect coverage improvements.

**Comment by danpetry** [comment]
Current automation status should be indicated by any scripts that are provided in the test folder.

I think it would be good nevertheless to point towards Jose's open PR in the top folder README, maybe in a section marked "helper scripts" or "resources" or something.

**Comment by danpetry** [comment]
Closing and opening two separate PRs (document clarification, filename change) and one issue (automation improvements).

File: README.md (hunk 5 above)
Comment (danpetry) [reply README.md]: changed

**Pull request closed without merging**